sdk/typespec/typespec_macros/README.md

```diff
@@ -0,0 +1,82 @@
+# TypeSpec Macros crate for Rust
+
+The TypeSpec Macros crate provides procedural macros for [TypeSpec](https://typespec.io)-generated client libraries. These macros simplify the implementation of common patterns required when working with TypeSpec-generated code.
```

Comment:
This PR does not include the README.md file in the `lib.rs` file as per the Rust SDK authoring guidelines. If it did, it would have caught the errors in the doccomment for `SafeDebug`.